### ISSUE TITLE
The instant-nsr-pl neus version of mesh is less effective than the native neus version? Is it a problem with my operation?

### ISSUE THREAD

**Issue by sunshineywz123** [bennyguo/instant-nsr-pl#9]
The instant-nsr-pl neus version of mesh is less effective than the native neus version? Is it a problem with my operation?
![image](https://user-images.githubusercontent.com/32233542/200324598-9350d3c3-4d38-4d35-b6be-576af75a6d90.png)
![image](https://user-images.githubusercontent.com/32233542/200324701-158c696b-784a-4b21-a94f-64dd7bd87e83.png)
![image](https://user-images.githubusercontent.com/32233542/200324712-b8c579ea-d483-4352-b277-88705ea2e65f.png)

**Comment by bennyguo** [comment]
It's the expected output with the default training setting. If you want higher-quality meshes, please try training for more iterations or increasing num_samples_per_ray (use the latest commit to avoid manual calculation of render_step_size, which is `2*sqrt(3)*radius/num_samples_per_ray`).

**Comment by SevenLJY** [comment]
Just a thought: is it possible that the lower-quality mesh is due to a different lr scheduler than the original NeuS? 

Now the implementation is an ExponetialLR, but it was a more complex scheduler with a warm-up stage. A slower lr at the beginning might guide a more soft surface deformation than the ExponetialLR.

**Comment by yyeboah** [comment]
Is it possible that this drop in surface reconstruction quality is not entirely due to sub-optimal hyperparameters but also in part, due to the interpolation that is adopted in these Occupancy Grid-based hybrid techniques?

**Comment by bennyguo** [comment]
@yyeboah I think this hybrid approach works fine since the grid-interpolated feature is followed by an MLP which brings non-linearity. It’s also demonstrated to be effective in several recent research works.

**Comment by yyeboah** [comment]
@bennyguo You do make a pretty strong point. Thanks for calling my attention to the MLP component, which for some reason, completely slipped my mind. 

By the way, cheers for working on this project. Can't wait to see the ripples it makes.

**Comment by bennyguo** [comment]
@yyeboah Thanks!

**Comment by bennyguo** [comment]
Please see https://github.com/bennyguo/instant-nsr-pl/issues/26 for my latest findings about improving quality and try our latest code :)

**Comment by bennyguo** [comment]
There was a bug in the code which degraded the quality of extracted meshes, which was fixed in 2cd3858f2ae85838a668a19de0f3e1402be2e022. Now the mesh quality should match the original NeuS implementation. @sunshineywz123 @yyeboah @SevenLJY
